client/src/components/SignUpAccountPage.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { accountPageSubmit, createUser } from '../actions/actions.js';
+
+export class SignUpAccountPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: null,
+      lastName: null,
+      phone: null,
+      password: null,
+      passwordVerify: null,
+      fieldErrors: false
+    };
+
+    this.onClickSubmit = this.onClickSubmit.bind(this);
+  }
+
+  onClickSubmit() {
+    // this.props.dispatch(accountPageSubmit());
+    // console.log(this.state);
+
+    if (this.state.firstName === null ||
+      this.state.lastName === null ||
+      this.state.phone === null ||
+      this.state.password === null ||
+      this.state.password !== this.state.passwordVerify) {
+      this.setState({
+        fieldErrors: true
+      })
+    } else {
+      console.log('fields ok')
+      var user = {
+        firstName: this.state.firstName,
+        lastName: this.state.lastName,
+        phone: this.state.phone,
+        email: this.props.user.email,
+        password: this.state.password
+      }
+      this.props.dispatch(createUser(user));
+    }
+
+  }
+
+  onChangeFirstName(firstName) {
+    this.setState({
+      firstName: firstName
+    });
+  }
+
+  onChangeLastName(lastName) {
+    this.setState({
+      lastName: lastName
+    });
+  }
+
+  onChangePhone(phone) {
+    this.setState({
+      phone: phone
+    });
+  }
+
+  onChangePassword(password) {
+    this.setState({
+      password: password
+    });
+  }
+
+  onChangePasswordVerify(password) {
+    this.setState({
+      passwordVerify: password
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <form className="ui form">
+          <h4 className="ui dividing header">Account Information</h4>
+          <div className="field">
+            <label>Name</label>
+            <div className="two fields">
+              <div className="field">
+                <input type="text" placeholder="First Name"
+                  onChange={(e) => {this.onChangeFirstName(e.target.value)}}/>
+              </div>
+              <div className="field">
+                <input type="text" placeholder="Last Name"
+                  onChange={(e) => {this.onChangeLastName(e.target.value)}}/>
+              </div>
+            </div>
+          </div>
+          <div className="field">
```

Comment:
A lot of this code is repeated, and could be replaced with a single block of code replicated using a map function